src/hotspot/os/linux/os_linux.cpp

```diff
@@ -5134,10 +5130,11 @@ static jlong slow_thread_cpu_time(Thread *thread, bool user_sys_cpu_time) {
                  &ldummy, &ldummy, &ldummy, &ldummy, &ldummy,
                  &user_time, &sys_time);
   if (count != 13) return -1;
+  double clock_tics_factor = 1000000000 / os::Posix::clock_tics_per_second();
```

Comment:
Why use `double` here? The right hand side is an `int` expression and the use of `clock_ticks_factor` below is in a `jlong` expression yielding a `jlong` function result. If you want to avoid integer rounding, you have to cast at least one res operand to `double`.